dbms/tests/queries/0_stateless/00458_merge_type_cast.reference

```diff
@@ -14,7 +14,8 @@
  1:
 1
 1
- -1: 
+ -1:
+18446744073709551615
```

Comment:
Previous test cases should be problematic. But I'm not sure.

Reply:
It is not correct, because the query was `SELECT x FROM test.merge_s64_u64 WHERE x IN (-1);`.

Reply:
@KochetovNicolai I need some help : )
My current idea is to rerun the filter stream. but the required columns in the where_expression may no longer exist. can I get some advice here ? [Code](https://github.com/yandex/ClickHouse/pull/3159/files#diff-e07f6e9213f722fdd54782f5934373aaR453)

Reply:
Maybe I need to push down the `where_expression_info` and remove_where_column ? Like `prewhere_expression_info` ?